avocado/core/dispatcher.py

```diff
@@ -51,10 +51,18 @@ def plugin_type(self):
         else:
             return self.namespace
 
+    def fully_qualified_name(self, extension):
```

Comment:
Also if we want to keep this in dispatcher itself, I think we should use a better names as it works for the extensions. Something like `get_extension_type` and `get_extension_fqn`. But I still think it'd be better to move this into the extensions viz previous comment.